__init__.py

```diff
@@ -61,93 +63,280 @@ def mall_link(self):
             self.speak_dialog('found_shop', {"name": shop['name'], "hours": hours, "location": location})
-            print({"name": shop['name'], "hours": hours, "location": location})
-            #self.gui.show_image(shop['logo'])
-        return 3, None
+            LOG.info({"name": shop['name'], "hours": hours, "location": location})
+            self.gui.show_image(shop['logo'], caption=f'{hours} {location}', title=shop['name'])
 
-    def more_than_one(self, shop_info):
-        self.speak_dialog('more_than_one')
-        speak_all_shops = self.ask_yesno('speak_all_shops')
-        if speak_all_shops == 'yes':
-            self.speak_dialog('all_locations')
-            return self.speak_shops(shop_info)
+    def location_selection(self, shop_info):
+        """
+        If there are several shops in found shops list
+        and user wants to get shop info on the certain
+        floor. If shop on that floor exists speaks
+        this shop info. Else speaks all shops info.
+        Args:
+            shop_info (list): found shops on user's
+                                request
+        Returns:
+            3, None (to ask for another shop info)
+        """
+        LOG.info(f"Shop by location selection {shop_info}")
+        floor = self.get_response('which_floor')
+        shops = shop_selection_by_floors(floor, shop_info)
+        if shops:
+            self.speak_shops(shops)
         else:
-            shop_by_floor = self.ask_yesno('shop_by_floor')
-            if shop_by_floor == 'yes':
-                floor = self.get_response('which_floor')
-                shop = self.request_handler.shop_selection_by_floors(floor, shop_info)
-                if shop is not None:
-                    return self.speak_shops([shop])
+            self.speak_dialog('no_shop_on_level')
+            self.speak_shops(shop_info)
+        return 3, None
+
+    def open_shops_search(self, shop_info, day_time, hour, min):
+        """
+       Selects open shops. Collects the list of
+       open shops else return empty list.
+       Args:
+           shop_info (list): found shops on user's
+                               request
+       Returns:
+           shop_info (list): open shops
+       """
+        open_shops = []
+        LOG.info(f"User's time {day_time, hour, min}")
+        for shop in shop_info:
+            parse_time = re.findall(r'(\d+)+[am|pm]', shop['hours'])
+            LOG.info(f'Parsed time {parse_time}')
+            open_time = int(parse_time[0])
+            close_time = int(parse_time[1])
+            if open_time <= hour < close_time:
+                open_shops.append(shop)
+            elif day_time[1] == 'am' and open_time <= hour:
+                open_shops.append(shop)
+        return open_shops
+
+
+    def time_calculation(self, shop_info, open, day_time, hour, min):
+        # add logic if shop opens and closes not at am-pm time period
+        # change to speak dialog
```

Comment:
Is this a TODO or just leftover comment? The docstring is good, so I think these lines can be removed